sqlmesh/core/plan/definition.py

```diff
@@ -66,60 +68,69 @@ class Plan:
         include_unmodified: Indicates whether to include unmodified models in the target development environment.
     """
 
+    context_diff: ContextDiff
+    restate_models: t.Optional[t.Iterable[str]] = None
+    no_gaps: bool = False
+    skip_backfill: bool = False
+    is_dev: bool = False
+    forward_only: bool = False
+    environment_ttl: t.Optional[str] = None
+    categorizer_config: CategorizerConfig = Field(default_factory=CategorizerConfig)
+    auto_categorization_enabled: bool = True
+    include_unmodified: bool = False
+
+    _directly_modified: t.List[Snapshot] = None
+    _indirectly_modified: SnapshotMapping = None
+
+    _plan_id: str = PrivateAttr(default_factory=random_id)
```

Comment:
why do you need privateattr?

Reply:
https://docs.pydantic.dev/latest/usage/models/#private-model-attributes

It needs a default factory so rather than use `Field`, `PrivateAttr` seems to be more idiomatic for pydantic here.